### PR TITLE
CASSANDRA-17039 Start checking for expired threads after the sync tasks they run have completed

### PR DESCRIPTION
**Pull request by maedhroz**

### CODE REVIEW

File: test/unit/org/apache/cassandra/repair/RepairJobTest.java [hunk 1]
```diff
@@ -274,9 +277,9 @@ public void testNoTreesRetainedAfterDifference() throws Throwable
         long millisUntilFreed;
         for (millisUntilFreed = 0; millisUntilFreed < TEST_TIMEOUT_S * 1000; millisUntilFreed += THREAD_TIMEOUT_MILLIS)
         {
-            // The measured size of the syncingTasks, and result of the computation should be much smaller
+            // Once the threads die, the size of the session should return to nearly pre-execution levels.
             TimeUnit.MILLISECONDS.sleep(THREAD_TIMEOUT_MILLIS);
-            if (ObjectSizes.measureDeep(session) < 0.8 * singleTreeSize)
+            if (ObjectSizes.measureDeep(session) <= (sizeBeforeTasks * 1.05))
```
Comment (maedhroz) [inline test/unit/org/apache/cassandra/repair/RepairJobTest.java]: The starting size of the session (before a single task is submitted) is much higher in Java 11 than Java 8 for some reason, but the amount of memory freed by the dying threads is almost exactly the same. The idea here is that we simply remember the starting size and make sure we revert to nearly that once the workers expire.

Comment (dcapwell) [reply test/unit/org/apache/cassandra/repair/RepairJobTest.java]: to help maintenance it would be good to document this

Comment (dcapwell) [reply test/unit/org/apache/cassandra/repair/RepairJobTest.java]: >  revert to nearly that once the workers expire.

what is the extra 5% for?

Comment (dcapwell) [reply test/unit/org/apache/cassandra/repair/RepairJobTest.java]: given how I read the test, this patch looks fine to me; just making sure we revert to where we were before starting; which is mostly what this test is checking....

+1 for that, but would still be nice to know what got larger.

Comment (maedhroz) [reply test/unit/org/apache/cassandra/repair/RepairJobTest.java]: I'll make a note around this in the Jira and expand the comment inline here as well.

**Review by dcapwell — APPROVED** [review]
+1

**Comment by maedhroz** [comment]
Committed as https://github.com/apache/cassandra/commit/092bb60ba413b8ef0eb9e0de86ce394a2f939084

**Pull request closed without merging**